android/src/main/java/com/revenuecat/purchases/hybridcommon/PaywallFragment.kt

```diff
@@ -2,16 +2,40 @@ package com.revenuecat.purchases.hybridcommon
 
 import android.os.Bundle
 import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentActivity
+import androidx.lifecycle.ViewModelProvider
 import com.revenuecat.purchases.ui.revenuecatui.ExperimentalPreviewRevenueCatUIPurchasesAPI
 import com.revenuecat.purchases.ui.revenuecatui.activity.PaywallActivityLauncher
 import com.revenuecat.purchases.ui.revenuecatui.activity.PaywallResult
 import com.revenuecat.purchases.ui.revenuecatui.activity.PaywallResultHandler
 
 @OptIn(ExperimentalPreviewRevenueCatUIPurchasesAPI::class)
-internal class PaywallFragment(
-    private val requiredEntitlementIdentifier: String?,
-) : Fragment(), PaywallResultHandler {
+internal class PaywallFragment : Fragment(), PaywallResultHandler {
```

Comment:
Note that since we're using a fragment, this does require the hybrid to use some subclass of a `FragmentActivity`... But most hybrids provide some of those, without having to provide our own version.

Reply:
Remember that we found out that we needed to use an activity on Flutter, as they say that's the preferred method. Are you thinking of forcing users to use our own fragment parent class?

Reply:
No, the idea is for the developer to use `FlutterFragmentActivity` (provided by flutter) instead of the `PurchasesFlutterActivity`. That way, the current `presentPaywallFromFragment` method we have here should take care of launching this fragment and from there the activity.

I believe the reason we created that custom activity was to try to get a result back from the paywall activity right? @NachoSoto

Reply:
That would be solved with the view model we're adding here